apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx

```diff
@@ -25,27 +42,53 @@ export function ExternalDappLoader({
 }: ExternalDappLoaderProps) {
   const { chainId } = useWalletInfo()
 
-  useEffect(() => {
-    let isRequestRelevant = true
+  const fetchManifest = useCallback(
+    async (url: string) => {
+      if (!url) return
+
+      setLoading(true)
+
+      try {
+        const validation = validateHookDappUrl(url)
+        if (!validation.isValid) {
+          setManifestError(validation.error)
+          setDappInfo(null)
+          setLoading(false)
+          return
+        }
+
+        const trimmedUrl = url.trim()
+        const manifestUrl = `${trimmedUrl}${trimmedUrl.endsWith('/') ? '' : '/'}manifest.json`
 
-    setLoading(true)
+        const response = await fetchWithTimeout(manifestUrl, { timeout: TIMEOUT })
+        if (!response.ok) {
+          setManifestError('Failed to fetch manifest. Please verify the URL and try again.')
```

Comment:
would it make sense to specify the URL that was attempted in the message? would that help to debug the issue?

Reply:
Specified the URL now